### PR TITLE
Pull the servicePort from environment variable if unavailable from command line

### PR DESCRIPTION
**Pull request by drcrallen**
Pulls from `PORT`
Fixes https://github.com/druid-io/druid/issues/1124

### CODE REVIEW

File: server/src/main/java/io/druid/server/DruidNode.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import com.fasterxml.jackson.annotation.JacksonInject;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.api.client.repackaged.com.google.common.base.Strings;
```
Comment (gianm) [inline server/src/main/java/io/druid/server/DruidNode.java]: Please use the real guava import.

File: server/src/test/java/io/druid/server/DruidNodeTest.java [hunk 2]
```diff
@@ -17,7 +17,10 @@
 
 package io.druid.server;
 
+import com.google.api.client.repackaged.com.google.common.base.Strings;
```
Comment (gianm) [inline server/src/test/java/io/druid/server/DruidNodeTest.java]: Ditto on the import.

File: server/src/main/java/io/druid/server/DruidNode.java (hunk 1 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/server/DruidNode.java]: sometimes intelliJ is silly

**Comment by gianm** [comment]
I'm not really a fan of hard-coding environment variables in Druid configuration classes, especially vague ones like PORT. I'd prefer asking people to write wrapper scripts that take $PORT and turn it into -Dio.druid.port="$PORT". We could provide scripts like this for our favorite execution environments if we want.

This is for a couple of reasons.

First, I think we have enough ways of specifying configs: the properties file for pre-distributed stuff and jvm system properties for injected-at-startup stuff. Other than some whackiness with the Hadoop extensions, that's also all tucked away in the PropertiesModule, meaning there's just one place in the code that static properties can originate from. I think that's good.

Second, the purpose of these sorts of environment variables is to make it easier to run Druid in some execution environment. Those environments aren't standardized, so I don't think the benefit of simplifying just one of them is worth the potential annoyance of having things like this hard-coded.

If we do go with this patch, it definitely needs to include documentation, and should probably be part of new docs section describing what environment variables Druid respects.

**Comment by drcrallen** [comment]
Given there are workarounds for this limitation, and the lack of a clear standard to respect, I'm closing this for the time being.

**Pull request closed without merging**